upf/upf_pfcp_api.c

```diff
@@ -694,7 +694,7 @@ handle_f_teid (upf_session_t * sx, upf_main_t * gtm, pfcp_pdi_t * pdi,
       if ((pdi->f_teid.flags & F_TEID_CHID))
 	{
 	  u8 chid = pdi->f_teid.choose_id;
-	  teid = sparse_vec_index (sx->teid_by_chid, chid);
+	  teid = *sparse_vec_elt_at_index (sx->teid_by_chid, chid);
 
 	  if (teid)
 	    chosen = true;
```

Comment:
* chosen is not required, teid can be used directly since choosing TEID = 0 is not valid
* it might be slightly clearer to use sparse_vec_index and test for the zero index instead of relying on the index zero element being 0

```suggestion
      if ((pdi->f_teid.flags & F_TEID_CHID))
	{
	  uword i = sparse_vec_index (sx->teid_by_chid, pdi->f_teid.choose_id);

	  if (i)
	    teid = vec_elt (sx->teid_by_chid, i);
	}
```

Reply:
Thanks, done